x-pack/plugins/infra/common/http_api/log_entries/highlights.ts

```diff
@@ -52,8 +52,8 @@ export type LogEntriesHighlightsRequest = rt.TypeOf<typeof logEntriesHighlightsR
 export const logEntriesHighlightsResponseRT = rt.type({
   data: rt.array(
     rt.type({
-      topCursor: logEntriesCursorRT,
-      bottomCursor: logEntriesCursorRT,
+      topCursor: rt.union([logEntriesCursorRT, rt.null]),
+      bottomCursor: rt.union([logEntriesCursorRT, rt.null]),
       entries: rt.array(logEntryRT),
     })
```

Comment:
What do you think about making this an all-or-nothing type by moving the union up as in

```typescript
rt.union([
  rt.type({
    topCursor: rt.null,
    bottomCursor: rt.null,
    entries: rt.array(logEntryRT),
  }),
  rt.type({
    topCursor: logEntriesCursorRT,
    bottomCursor: logEntriesCursorRT,
    entries: rt.array(logEntryRT),
  }),
])
```

?

(I would have proposed to derive a branded `EmptyArray` runtime type for the `entries` in the first case, but that felt a bit excessive :innocent:)

Reply:
It's easy enough to make, and it better represents the truth. I'll change it